splinterd/src/main.rs

```diff
@@ -247,20 +242,22 @@ fn start_daemon(matches: ArgMatches) -> Result<(), UserError> {
         }
     };
 
+    let local_node_registry_location = format!("{}{}", config.state_dir(), "nodes.yaml");
+
     let rest_api_endpoint = config.bind();
 
     #[cfg(feature = "database")]
     let db_url = config.database();
 
-    let registry_backend = config.registry_backend();
-
     let admin_service_coordinator_timeout = config.admin_service_coordinator_timeout();
 
     config.log_as_debug();
 
+    #[allow(unused_mut)]
```

Comment:
A much better approach is to use it always mutable.

```
let mut daemon_builder = SplinterDaemonBuilder::new();

daemon_builder
    .with_storage_location(storage_location)
    .with_key_registry_location(key_registry_location)
    ...
```

Reply:
Updated